### PR TITLE
DOC: Update to clarify actual behavior real_if_(all elements)_close

### PR DESCRIPTION
**Pull request by ghost**
Related to #15626

 - Updated the description to consider all array elements
 - Updated the examples to use multiple elements array, to show that one element not close enough prevent for the whole array to be considered as real

### CODE REVIEW

File: numpy/lib/type_check.py [hunk 1]
```diff
@@ -492,7 +492,8 @@ def _real_if_close_dispatcher(a, tol=None):
 @array_function_dispatch(_real_if_close_dispatcher)
 def real_if_close(a, tol=100):
     """
-    If complex input returns a real array if complex parts are close to zero.
+    If complex input returns a real array if all complex parts of the 
+    input array are close to zero.
```
Comment (rossbar) [inline numpy/lib/type_check.py]: Consider removing the word "array" or replacing with "array-like".

File: numpy/lib/type_check.py [hunk 2]
```diff
@@ -492,7 +492,8 @@ def _real_if_close_dispatcher(a, tol=None):
 @array_function_dispatch(_real_if_close_dispatcher)
 def real_if_close(a, tol=100):
     """
-    If complex input returns a real array if complex parts are close to zero.
+    If complex input returns a real array if all complex parts of the
```
Comment (rossbar) [inline numpy/lib/type_check.py]: Might as well take the opportunity to clean up the grammar here a bit as well: consider replacing "If complex input" with "If input is complex,"

**Review by rossbar — COMMENTED** [review]
Generally LGTM, just a couple comments on wording.

As @seberg suggested, it might be worthwhile to "gently discourage" use of this function in the doc string. For example, you could add a `.. note` before the `Parameters` section starts that indicates this is a convenience function that is equivalent to (less the implicit `else`)
```python
tol = 100 * np.finfo(a.real.dtype).eps
if np.all( np.abs(np.imag(a)) < tol ):
    out = a.real
```
and that being explicit is generally preferred.

I don't feel strongly about this and I think the changes you made do improve clarity, so I'm +1 for merge (after addressing the inline comments)

**Comment by and-sang** [comment]
If it's alright, I will try updating the tutorial as a mean to discourage its use, instead of adding a note to the function.

**Comment by eric-wieser** [comment]
@rossbar: perhaps a spelling using `np.allclose` or `np.isclose` would be better there.

File: numpy/lib/type_check.py [hunk 3]
```diff
@@ -492,7 +492,8 @@ def _real_if_close_dispatcher(a, tol=None):
 @array_function_dispatch(_real_if_close_dispatcher)
 def real_if_close(a, tol=100):
     """
-    If complex input returns a real array if complex parts are close to zero.
+    If input is complex with all complex parts close to zero, return real
```
Comment (rossbar) [inline numpy/lib/type_check.py]: General approach to rewording seems fine, but the second "complex" should be "imaginary", i.e.

> If input is complex with all imaginary parts close to zero, return real parts.

**Review by rossbar — CHANGES_REQUESTED** [review]

**Comment by rossbar** [comment]
Good suggestion, though I think @and-sang 's idea of submitting a PR to the original tutorial is more important.

I'm still ambivalent about adding a `.. note` to the docs to discourage use, but if you do decide to do so, definitely take @eric-wieser 's comment into account to make my proposed example less busy.

**Review by rossbar — APPROVED** [review]
LGTM

**Comment by rossbar** [comment]
FWIW the CI errors appear to be unrelated (looks like a test of einsum is failing for PyPy3)

**Comment by seberg** [comment]
Thanks @and-sang and @rossbar for the review, putting this in then. Did you want to followup on mentioning `np.allclose` or so?

**Comment by rossbar** [comment]
@seberg  the `np.allclose` comment was related to my suggestion for adding a `.. note` to the docstring to discourage the use of `np.real_if_close`. I had originally suggested something like

```python
tol = 100 * np.finfo(a.real.dtype).eps
if np.all( np.abs(np.imag(a)) < tol ):
    out = a.real
```
Which @eric-wieser (correctly) pointed out could be written much more clearly and succinctly with `np.allclose`. If we're not adding a "discouragement" note to the docs, I think everything is square!

**Pull request merged**